Daniel Cogan

danield9tqh opened the pull request.

Commit:
Finished Description and Plan

emeersman committed:
Finished initial critique for project description

Commit:
Added Design and Implementation

bwiedermann commented on documents/description.md:
```diff
@@ -1,9 +1,40 @@
+tuples, lack most of these very useful functions. This can result in many over-
+complicated SQL queries. The goal of this project is to augment the experience 
+of working with and manipulating SQL data by providing the above mentioned 
+Collection functions for SQL databases.
```
Quite a bit more detail here could be helpful, especially for the uninitiated. For instance, it could be useful to include a specific example of a query that could be improved with functional features.

bwiedermann commented on documents/plan.md:
In general, you should include more detail in your future milestones.

bwiedermann commented on documents/plan.md:
```diff
@@ -1,11 +1,38 @@
+implement in my language. I plan to use one of the Scala testing frameworks we
+discussed in class to make sure that my function implementations actually return
+the correct results. Because my project fits very well with unit testing (each
+function has a test), I will probably practice test-driven development. 
```
probably? :confused:

bwiedermann commented on documents/description.md:
```diff
@@ -1,9 +1,40 @@
+databases in Scala. The one most similar to this project is [Slick](http://slick.typesafe.com). Slick provides
+many of the given Collection functions for SQL databases and will probably have
+a big influence on this project. [Here](http://manuel.bernhardt.io/2014/02/04/a-quick-tour-of-relational-database-access-with-scala/) is another article detailing other options
+for working with SQL in Scala.
```
In what ways will your project differ from Slick (if at all)?

rjustice05 committed:
added Critique for April 5

Commit:
Implemented basic filter expression.

This implementation includes:
- A basic SQL/Relational Algebra Abstract syntax
- Conversion from the Abstract syntax to a more readable SQL query. The
SQL query however is not yet syntactically correct.
- A basic implementation of a Scala collection
- Initial method for users to define a DB table schema

Commit:
Fixed package structure

Commit:
Implemented translation to MySQL

emeersman committed:
Updating third critique

Commit:
Removed project code from this repo

rjustice05 committed:
Completed Critique for April 19

Commit:
Finished Final Documentation
